pkg/gardenlet/controller/shoot/shoot/reconciler_force_delete.go

```diff
@@ -182,6 +182,11 @@ func (r *Reconciler) runForceDeleteShootFlow(ctx context.Context, log logr.Logge
 			Fn:           flow.TaskFn(botanist.DeleteSeedNamespace).RetryUntilTimeout(defaultInterval, defaultTimeout),
 			Dependencies: flow.NewTaskIDs(syncPoint, waitUntilEtcdsDeleted, deleteKubernetesResources),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Delete public service account signing keys from Garden cluster",
+			Fn:           botanist.DeletePublicServiceAccountKeys,
+			Dependencies: flow.NewTaskIDs(syncPoint),
+		})
```

Comment:
Can we add `ownerReferences` to the `Secret` and remove this explicit deletion from here?

Reply:
I do not think that this is possible. Please see [the discussion](https://github.com/gardener/gardener/pull/9024#discussion_r1479397229) we had some weeks ago.